src/inmanta/deploy/work.py

```diff
@@ -85,6 +90,9 @@ class TaskQueueItem:
     task: PrioritizedTask[Task]
     insert_order: int
 
+    # Mutable state
+    is_active: bool = True
```

Comment:
I moved this here (instead of a set) for simplicity, I hope I didn't violate any delicate mechanism